src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -895,13 +895,33 @@ private BoundCollectionExpression ConvertCollectionExpression(
                 }
 
                 implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
-                collectionCreation = BindCollectionExpressionConstructor(syntax, targetType, constructor, diagnostics);
-                Debug.Assert((collectionCreation is BoundNewT && !isExpanded && constructor is null) ||
-                             (collectionCreation is BoundObjectCreationExpression creation && creation.Expanded == isExpanded && creation.Constructor == constructor));
 
-                if (collectionCreation.HasErrors)
+                // Bind collection creation with arguments.
+                foreach (var element in elements)
```

Comment:
It looks like we're converting the arguments in the with-element even when the with-element is not in the right position, or there are multiple. Feels like we should only convert the one that is in the right position, and do some error binding (bind to natural type?) for the error ones.

Reply:
It seems useful to me to bind `with(...)` as arguments to the best candidate method regardless of whether the arguments are in the first position, since presumably that is the intent.

Reply:
This still makes me uncomfortable. 
Two reasons:
1. Semantically, there's only one object creation in a collection expression. But presumably QuickInfo on each `with` might show different constructors...
2. When there are multiple `with` elements, any diagnostics reported on those in the wrong position are guaranteed to be cascading diagnostics. Do we have a test for that?
It may be good to get a second opinion and/or discuss after standup.

Reply:
> When there are multiple with elements, any diagnostics reported on those in the wrong position are guaranteed to be cascading diagnostics.

Overload resolution for the constructor is independent of the position of the `with()`, so one does not affect the other.

> Do we have a test for that?

Yes, `CollectionExpressionArguments.LanguageVersion_03()` tests `List<int> l = [with(x: 1), with(y: 2)];`

Reply:
Added a PROTOTYPE comment for now, to revisit this later, likely when implementing `IOperation` support.